source/common/json/json_loader.cc

```diff
@@ -521,7 +523,7 @@ bool ObjectHandler::StartObject() {
     state_ = expectKeyOrEndObject;
     return true;
   default:
```

Comment:
the idea of NOT_REACHED is to make the compiler happy in cases we know something will not happen rather than detect cases like that.

In this case, it feels more like InvalidStateException or something like that (i do not think we really have a variety of exception types in our code base), so throwing Json::Exception might make sense, and specify state_ as part of the exception message.